python/vowpalwabbit/pyvw.py

```diff
@@ -40,27 +61,109 @@ def run():
 
     def learn(self, data_iterator):
-        """Train search task by providing an iterator of examples"""
+        """Train search task by providing an iterator of examples.
+
+        Parameters
+        ----------
+
+        data_iterator: iterable objects
+            Consists of examples to be learned
+
+        Returns
+        -------
+
+        self : SearchTask
+
+        """
         for my_example in data_iterator.__iter__():
             self._call_vw(my_example, isTest=False)
 
     def example(self, initStringOrDict=None, labelType=pylibvw.vw.lDefault):
-        """Create an example
-            initStringOrDict can specify example as VW formatted string, or a dictionary
-            labelType can specify the desire label type"""
+        """Create an example initStringOrDict can specify example as VW
+        formatted string, or a dictionary labelType can specify the desire
+        label type
+
+        Parameters
+        ----------
+
+        initStringOrDict : str/dict
+            Example in either string or dictionary form
+        labelType : integer
+            - 0 : lDEFAULT
+            - 1 : lBINARY
+            - 2 : lMULTICLASS
+            - 3 : lCOST_SENSITIVE
+            - 4 : lCONTEXTUAL_BANDIT
+            - 5 : lMAX
+            - 6 : lCONDITIONAL_CONTEXTUAL_BANDIT
+            The integer is used to map the corresponding labelType using the
+            above available options
+
+        Returns
+        -------
+
+        out : Example
+
+        """
         if self.sch.predict_needs_example():
             return self.vw.example(initStringOrDict, labelType)
         else:
             return self.vw.example(None, labelType)
 
     def predict(self, my_example, useOracle=False):
-        """Return prediction"""
+        """Predict on the example
+
+        Parameters
+        ----------
+
+        my_example : Example
+            example used for prediction
+        useOracle : bool
+
+        Returns
+        -------
+
+        out : integer
+            Prediction on the example
+        """
         self._call_vw(my_example, isTest=True, useOracle=useOracle)
         return self._output
 
 
 def get_prediction(ec, prediction_type):
-    """Get specified type of prediction from example"""
+    """Get specified type of prediction from example
+
+    Parameters
+    ----------
+
+    ec : Example
+    prediction_type
+        - pSCALAR : Scalar prediction type
+        - pSCALARS : Multiple scalar-valued prediction type
+        - pACTION_SCORES : Multiple action scores prediction type
+        - pACTION_PROBS : Multiple action probabilities prediction type
+        - pMULTICLASS : Multiclass prediction type
+        - pMULTILABELS : Multilabel prediction type
+        - pPROB : Probability prediction type
+        - pMULTICLASSPROBS : Multiclass probabilities prediction type
+        - pDECISION_SCORES : Decision scores prediction type
+
```

Comment:
this is inconsistent with how we define label_type inputs, should it be integer type and then list the integers and corresponding enum names? the short descriptions probably aren't needed

Reply:
Okay, I will change it.